src/libproc_macro/lib.rs

```diff
@@ -200,6 +201,78 @@ impl Span {
     pub fn call_site() -> Span {
         ::__internal::with_sess(|(_, mark)| Span(mark.expn_info().unwrap().call_site))
     }
+
+    /// The original source file into which this span points.
+    #[unstable(feature = "proc_macro", issue = "38356")]
+    pub fn source_file(&self) -> Option<SourceFile> {
+        // If this span's token was produced by a macro
+        if self.0.ctxt != SyntaxContext::empty() {
+            return None
+        }
```

Comment:
Is it somehow possible to traverse up the macro call hierarchy? To get the span of the macro-invocation that generated the `self` span? Does this question make sense? :confused:

Reply:
It is possible, yes.

Reply:
So should it be done here? If not, it should be possible to do the recursive traversal in user space IMO (which is not possible right now, right?).

Reply:
I'm still not sure what you mean. There's already `Span::call_site()` which gets the span of the current macro invocation. I don't think we get much out of giving the macro call information about other macros than itself.

Reply:
Oh sorry, I missed `call_site()`. That's exactly what I meant! :)